### PR TITLE
"Prettier" with a new context for fetch options

### PR DESCRIPTION
**Pull request by birkan9977**
In this prettier version, a new context for fetch options is introduced both to see whether different contexts can work together and to enable manually submitting the API requests while maintaining an option for automatic submit on change events. The challenge was to update context every time an option is selected by the user while blocking data fetch until a manual submit takes place. React updates all components using that specific context from parents to children when a change occurs in context. New rules for preventing to send the same API requests were also introduced in this version. The code was formatted with "prettier" giving the name to this version...
Screen Shot 2020-08-31 at 17 57 14

### CODE REVIEW

File: src/components/data.jsx [hunk 1]
```diff
@@ -1,177 +1,210 @@
...
+        {data
+          ? data.map((item, index) => (
+              <>
+                <div id="repo-items">
+                  <div id="repo-list-items">
+                    <div id="order-number">{ItemIncrement()}</div>
+
+                    <li
+                      id="repo-list-items-name"
+                      key={`repo-list-items-name-${index}`}
```
Comment (vbilici) [inline src/components/data.jsx]: hmm I think there is a misunderstanding about keys. Basically you don't need to use `key` here. Check here for more information -> https://reactjs.org/docs/lists-and-keys.html

Comment (vbilici) [reply src/components/data.jsx]: keys are only needed if there is a list (arrays, etc...)

File: src/components/urlmaker.jsx [hunk 2]
```diff
@@ -1,36 +1,23 @@
...
-      } else {
-        filterLanguage = `language:${filters.language}`
-
-      }
-
-      return filterLanguage
+export default function urlMaker(filters) {
+  function languageCorrect() {
+    let filterLanguage = '';
+    if (filters.language === 'C++') {
```
Comment (vbilici) [inline src/components/urlmaker.jsx]: `switch` instead of `if` would be a better fit here

**Review by vbilici — CHANGES_REQUESTED** [review]

**Comment by birkan9977** [comment]
Thnks bro, i m on it😇

**Pull request merged**